tyr-runner/src/main/java/org/jboss/tyr/check/DescriptionCheck.java

```diff
@@ -48,6 +52,19 @@ public String check(JsonObject payload) {
                         break;
                     }
                 }
+                for (OptionalPatterns optionalRow : optionalRows){
+                    Matcher preconditionMatcher = optionalRow.getPrecondition().matcher(line);
+                    if (preconditionMatcher.matches()){
+                        Matcher patternMatcher = optionalRow.getPattern().matcher(line);
+                        if (!patternMatcher.matches()){
+                            return optionalRow.getMessage();
+                        }
+                        else {
+                            optionalRows.remove(optionalRow);
```

Comment:
Does it make sense to remove optional checks? Or is use case for using one optional check multiple times valid? Technically, same situation happens in case of required checks,

Reply:
yes, this is the concept. Only one occurrence is required for the required row. For optional it doesn't matter. See https://github.com/jboss/tyr/pull/139/files/07db5f9a2d70f81956d81b65fbebdac61cfd3fe6#diff-af594769d38bf7c5dd4aecf304b82eabed030e9fdaeb385b7df6d64f809b4af4R71

Reply:
actually removal of optional check is wrong because we should check all lines that pass precondition. But I won't block the PR on this.